src/main/java/games/strategy/engine/lobby/client/login/LobbyLogin.java

```diff
@@ -164,7 +164,8 @@ public void notifyFailedLogin(final String message) {
               final Map<String, String> props = new HashMap<>();
               props.put(LobbyLoginValidator.REGISTER_NEW_USER_KEY, Boolean.TRUE.toString());
               props.put(LobbyLoginValidator.EMAIL_KEY, createAccount.getEmail());
-              // TODO: Don't send the md5-hashed password once the lobby is updated
+              // TODO: Don't send the md5-hashed password once the lobby removes the support, kept for
```

Comment:
Recommend you make this TODO stronger. First suggestion is to indicate the conditions when we can remove this code, when will we no longer need the backward compatibility. From this context the answer is actually not obvious, from the PR and branch we know it is when we have clients sending bcrypted passwords, but from this code snippet you would not that.

Even better than that, I would recommend using a code branch with flag to allow us to work on this TODO with compile time checks. to do that you would create a file like `PasswordHashMigration`, with a single public static flag: `if(PasswordHashMigration.enableBackwardCompatibility)`. This way you can add all of the TODO commentary in one place to describe when the password migration can be updated to no longer support legacy. Also when we remove the backward compatibiliyt, we simply delete the flag and file and then fix any compilation errors. You could get the same type of location flagging by adding in a project key to the TODO comments, though the compile time version is a stronger and more automated version of that.